src/NServiceBus.Core.Tests/Routing/UnicastRoutingTableTests.cs

```diff
@@ -0,0 +1,97 @@
+﻿namespace NServiceBus.Core.Tests.Routing
+{
+    using System;
+    using System.Threading.Tasks;
+    using Extensibility;
+    using NServiceBus.Routing;
+    using NUnit.Framework;
+
+    [TestFixture]
+    class UnicastRoutingTableTests
+    {
+        [Test]
+        public void When_registering_multiple_static_routes_for_same_type_should_throw_exception()
+        {
+            var routingTable = new UnicastRoutingTable();
+            var firstRoute = UnicastRoute.CreateFromPhysicalAddress("address@somewhere");
+            var secondRoute = UnicastRoute.CreateFromEndpointName("sales");
+
+            routingTable.RouteTo(typeof(Command), firstRoute);
+
+            Assert.That(() => routingTable.RouteTo(typeof(Command), secondRoute), Throws.Exception.TypeOf<Exception>().And.Message.Contains($"The static routing table already contains a route for message '{nameof(Command)}'. Remove the ambiguous route registrations or override the existing route."));
+        }
+
+        [Test]
+        public async Task When_overriding_static_routes_with_same_type_should_only_use_last_route()
```

Comment:
last is a bit ambiguous. How about

`..._should_use_route_which_overrides_previous_routes` or something like that?